src/main/java/com/igot/cb/discussion/service/impl/DiscussionServiceImpl.java

```diff
@@ -1254,6 +1254,12 @@ public ApiResponse searchDiscussionByCommunity(Map<String, Object> searchData) {
             Map<String,Object> filterCriteria = new HashMap<>();
             filterCriteria.put(Constants.COMMUNITY_ID, searchData.get(Constants.COMMUNITY_ID));
             filterCriteria.put(Constants.TYPE, Constants.QUESTION);
+            if (cbServerProperties.isDiscussionReportHidePost()){
+                filterCriteria.put(Constants.STATUS,Arrays.asList(Constants.ACTIVE,Constants.REPORTED));
+            }else {
+                filterCriteria.put(Constants.STATUS,Arrays.asList(Constants.ACTIVE,Constants.REPORTED,Constants.SUSPENDED));
+            }
+            filterCriteria.put(Constants.IS_ACTIVE, true);
```

Comment:
is this code correct?
if the flag is set to true -- then you need to suspend based after checking the number of reports attribute... 
else nothing to do...